commit 6e8a8e0629dcc4c33926210fc107bf5cd68cf418
Author: Trent Mick <trentm@gmail.com>
Date:   2019-07-16T16:44:55-07:00 (3 months ago)
    
    TRITON-1818 explicitly specify PEM key format for keys generated by ssh-keygen
    Reviewed by: Angela Fong <angela.fong@joyent.com>
    Approved by: Angela Fong <angela.fong@joyent.com>

diff --git a/README.md b/README.md
index 0670e93..1a69ca3 100644
--- a/README.md
+++ b/README.md
@@ -95,7 +95,7 @@ and add an SSH key for access.
 2. Create a test user (we'll use "jill"):
 
         # On your dev machine, create a key
-        ssh-keygen -t rsa -f ~/.ssh/sdc-docker-jill.id_rsa -b 2048 -N ""
+        ssh-keygen -t rsa -m PEM -f ~/.ssh/sdc-docker-jill.id_rsa -b 2048 -N ""
 
         # Copy it to COAL so we can add it to the 'jill' account.
         scp ~/.ssh/sdc-docker-jill.id_rsa.pub root@10.99.99.7:/var/tmp/
diff --git a/test/integration/helpers.js b/test/integration/helpers.js
index c4359c9..a136fd9 100644
--- a/test/integration/helpers.js
+++ b/test/integration/helpers.js
@@ -712,7 +712,8 @@ GzDockerEnv.prototype.init = function denvInit(t, state_, cb) {
             p('# Creating "%s" SSH key for "%s" account', keyName, self.login);
             var cmds = [
                 fmt('rm -rf %s %s', self.privKeyPath, self.pubKeyPath),
-                fmt('ssh-keygen -t rsa -f %s -b 2048 -N ""', self.privKeyPath),
+                fmt('ssh-keygen -t rsa -m PEM -f %s -b 2048 -N ""',
+                    self.privKeyPath),
                 fmt('cp %s '
                     + '/zones/$(vmadm lookup -1 alias=sdc0)/root/var/tmp/',
                     self.pubKeyPath)
@@ -937,7 +938,8 @@ LocalDockerEnv.prototype.init = function ldenvInit(t, state_, cb) {
             p('# Creating "%s" SSH key for "%s" account', keyName, self.login);
             var cmds = [
                 fmt('rm -rf %s %s', self.privKeyPath, self.pubKeyPath),
-                fmt('ssh-keygen -t rsa -f %s -b 2048 -N ""', self.privKeyPath),
+                fmt('ssh-keygen -t rsa -m PEM -f %s -b 2048 -N ""',
+                    self.privKeyPath),
                 fmt('scp %s %s:/zones/%s/root/var/tmp/',
                     self.pubKeyPath, state.headnodeSsh, state.sdcZonename)
             ];
